Assets/Scripts/Player.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Player : MonoBehaviour
+{
+    public Rigidbody2D body;
+    public SpriteRenderer sprite;
+    public Animator animator;
```

Comment:
If this is the Animator Component of the Player GameObject itself you want to reference, you don't need to do this that way. Instead, you can simply use `GetComponent<Animator>()` to get it. same thing to SpriterRenderer I thing.

Reply:
Yep it is. I did consider doing it like that but I feel like its better to go for that way of doing things as it allows for easier de-coupling of things, the script could be used on a Player with no animator for example, we would only need to check that its not null and do not try to animate in that case. What do you think?

Reply:
I often see `if(GetComponent<Animator>())` to do this kind of test, so it doesnt change much. But I understand your feeling so do as you want, like I said, pretty much the same thing

Reply:
I often see if(GetComponent()) to do this kind of test, so it doesnt change much. But I understand your feeling so do as you want, like I said, pretty much the same thing

Reply:
Yep, those are two valid approach, without any big pro or cons for one or the other, I suggest we keep to this one, it allows us to change some behavior without modifying code (simply setting animator to null will disable animation for example) where the other approach requires you to remove the component